operator/v1/types_ingress.go

```diff
@@ -1346,6 +1385,59 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// headersManipulation specifies options for set/replace/append/delete headers.
+	HeadersManipulation IngressControllerHeadersManipulation `json:"headersManipulation,omitempty"`
+}
+
+// IngressControllerHeadersManipulation specifies options for set/replace/append/delete headers.
+type IngressControllerHeadersManipulation struct {
+	Response []IngressControllerSetReplaceAppendDeleteHTTPHeaders `json:"response,omitempty"`
+}
+
+// IngressControllerSetReplaceAppendDeleteHTTPHeaders specifies fields to mention the headers and the action you want to perform on them.
+// +union
+type IngressControllerSetReplaceAppendDeleteHTTPHeaders struct {
```

Comment:
I would suggest to change the name to something more simple like `IngressControllerHTTPHeadersActions`